internal/exec/exec.go

```diff
@@ -0,0 +1,530 @@
+// Package exec implements a minimalized external process launcher. It exists to work around some shortcomings for
+// Windows scenarios that aren't exposed via the os/exec package.
+package exec
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+	"syscall"
+	"unicode/utf16"
+	"unsafe"
+
+	"golang.org/x/sys/windows"
+)
+
+var (
+	errProcNotStarted  = errors.New("process has not started yet")
+	errProcNotFinished = errors.New("process has not finished yet")
+)
+
+// Exec is an object that represents an external process. A user should NOT initialize one manually and instead should
+// call New() and pass in the relevant options to retrieve one.
+//
+// The Exec object is not intended to be used across threads and most methods should only be called once per object.
+// It's expected to follow one of two conventions for starting and managing the lifetime of the process.
+//
+// Either: New() -> e.Start() -> e.Wait()
+//
+// or: New() -> e.Run()
```

Comment:
We should add something about ExitCode and Pid, etc I think

Reply:
Done!